flink-runtime/src/main/java/org/apache/flink/runtime/util/bash/BashJavaUtils.java

```diff
@@ -78,7 +78,7 @@ private static Configuration getConfigurationForStandaloneTaskManagers(String[]
 	private static void getJmResourceParams(String[] args) throws Exception {
 		JobManagerProcessSpec jobManagerProcessSpec = JobManagerProcessUtils.processSpecFromConfigWithFallbackForLegacyHeap(
```

Comment:
`processSpecFromConfigWithFallbackForLegacyHeap` sounds as if we are using another option if the legacy option has not been set. But isn't it actually the case that the provided option will be overridden by the legacy option if it is set? We might update the name of this method to better reflect this.

Reply:
alright, I added a hotfix renaming it to `processSpecFromConfigWithNewOptionToInterpretLegacyHeap(.., newOptionToInterpretLegacyHeap)`